websubhub-ballerina/annotation.bal

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Configuration for a WebSub Hub service.
+#
+# + leaseSeconds - The period for which the subscription is expected to be active in the Hub
+# + httpClientConfig - HTTP client configurations for subscription / unsubscription intent verification
+public type ServiceConfiguration record {|
+    int leaseSeconds?;
+    ClientConfiguration httpClientConfig?;
```

Comment:
In websub module, there is a similar configuration which is used for subscriber client. The field is named as `httpConfig`. Here, we have the field name as `httpClientConfig`. I prefer the both to be named as `clientConfig`. That is the convention we followed in security modules. WDYT?

Reply:
Yes it looks better. @shafreenAnfar WDYT?

Reply:
Yeah, there is a miss match and I agree the names we came up aren't the best. But we can't change it now. For this thing though, I am thinking something like `webHookConfig`

```ballerina
@websubhub:ServiceConfig {
    leaseSeconds: 86400,
    webHookConfig: {
        secureSocket: {
            cert: {
                path: "tests/resources/ballerinaTruststore.pkcs12",
                password: "ballerina"
            }
        }
    }
}
```

Reply:
In the other case it should have been something like 

```ballerina
record {|
        string|string[] url;
        http:ClientConfiguration ClientConfig?;
|} target;
```

Reply:
@ayeshLK @ldclakmal

Reply:
Yes. +1 for the suggested configurations.

Reply:
Also, I am not sure if all the configuration in http:ClientConfiguration is applicable here.

Reply:
* Updated `websubhub:ClientConfiguration` is as follows.
```ballerina
public type ClientConfiguration record {|
    string httpVersion = HTTP_1_1;
    http:ClientHttp1Settings http1Settings = {};
    http:ClientHttp2Settings http2Settings = {};
    decimal timeout = 60;
    http:PoolConfiguration? poolConfig = ();
    http:RetryConfig? retryConfig = ();
    http:ResponseLimitConfigs responseLimits = {};
    http:ClientSecureSocket? secureSocket = ();
|};
```
* Updated `websubhub:ServiceConfig` annotation usage is as follows.
```ballerina
@websubhub:ServiceConfig {
    webHookConfig: {
        secureSocket: {
            cert: {
                path: "tests/resources/ballerinaTruststore.pkcs12",
                password: "ballerina"
            }
        }
    }
}
```